From 4ddff6f42d3f7e905996e9c8201a0b5fbb461a50 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 3 May 2023 14:32:13 +0200
Subject: [PATCH] arm64: dts: st: add thermal zone description on stm32mp25

/// New commit message

Add thermal zone description on STM32MP25.

Rules established by STMicroelectronics:
-Tcritical = Tj - (sensor tolerance). With Tj the function temperature.
-Tlimitation = 90% of Tcritical.

On STM32MP25, Tj=125C even if the overdrive is used. So we have:
- STM32MP2xxA/C/D/F, Tcritical = 125-3 = 122C
- STM32MP2xxA/C/D/F, Tlimitation = 110C

Signed-off-by: Pascal Paillet <p.paillet@st.com>
Signed-off-by: Valentin Caron <valentin.caron@foss.st.com>
Change-Id: I545690800ca0480098f74bc8aecc8842417eb7b0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/390965
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 44 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/st/stm32mp253.dtsi |  4 +++
 2 files changed, 48 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 8584ec66bd98..667a3fdd47f0 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -24,6 +24,7 @@ cpu0: cpu@0 {
 			enable-method = "psci";
 			clocks = <&scmi_perf 0>;
 			clock-names = "cpu";
+			#cooling-cells = <2>;
 		};
 	};
 
@@ -150,6 +151,49 @@ psci {
 		method = "smc";
 	};
 
+	thermal-zones {
+		cpu-thermal {
+			polling-delay-passive = <250>;
+			polling-delay = <1000>;
+			thermal-sensors = <&dts 1>;
+
+			trips {
+				cpu_alert: cpu-alert {
+					temperature = <110000>;
+					hysteresis = <10000>;
+					type = "passive";
+				};
+
+				cpu-crit {
+					temperature = <122000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+			};
+
+			cooling-maps {
+				map0 {
+					trip = <&cpu_alert>;
+					cooling-device = <&cpu0 1 1>;
+				};
+			};
+		};
+
+		soc-thermal {
+			polling-delay-passive = <250>;
+			polling-delay = <1000>;
+			thermal-sensors = <&dts 0>;
+
+			trips {
+				soc-crit {
+					temperature = <122000>;
+					hysteresis = <1000>;
+					type = "critical";
+				};
+			};
+		};
+	};
+
 	timer {
 		compatible = "arm,armv8-timer";
 		interrupt-parent = <&intc>;
diff --git a/arch/arm64/boot/dts/st/stm32mp253.dtsi b/arch/arm64/boot/dts/st/stm32mp253.dtsi
index ef5a4facd470..c47bb9a89d9b 100644
--- a/arch/arm64/boot/dts/st/stm32mp253.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp253.dtsi
@@ -24,6 +24,10 @@ arm-pmu {
 	};
 };
 
+&optee {
+	interrupts = <GIC_PPI 15 (GIC_CPU_MASK_SIMPLE(2) | IRQ_TYPE_LEVEL_LOW)>;
+};
+
 &rifsc {
 	eth2: eth2@482d0000 {
 		compatible = "st,stm32mp25-dwmac", "snps,dwmac-5.10a";
-- 
2.39.5

